src/Event/Recommendations/RecommendationsForEnrichedOfferRepository.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Event\Recommendations;
+
+use CultuurNet\UDB3\Iri\IriGeneratorInterface;
+use CultuurNet\UDB3\ReadModel\DocumentRepository;
+use CultuurNet\UDB3\ReadModel\DocumentRepositoryDecorator;
+use CultuurNet\UDB3\ReadModel\JsonDocument;
+
+final class RecommendationsForEnrichedOfferRepository extends DocumentRepositoryDecorator
+{
+    private RecommendationsRepository $recommendationsRepository;
+
+    private IriGeneratorInterface $iriGenerator;
+
+    public function __construct(
+        RecommendationsRepository $recommendationsRepository,
+        IriGeneratorInterface $iriGenerator,
+        DocumentRepository $documentRepository
+    ) {
+        parent::__construct($documentRepository);
+        $this->recommendationsRepository = $recommendationsRepository;
+        $this->iriGenerator = $iriGenerator;
+    }
+
+    public function fetch(string $id, bool $includeMetadata = false): JsonDocument
+    {
+        $jsonDocument = parent::fetch($id, $includeMetadata);
+
+        if ($includeMetadata) {
+            $jsonDocument = $this->enrich($jsonDocument);
+        }
+
+        return $jsonDocument;
+    }
+
+    private function enrich(JsonDocument $jsonDocument): JsonDocument
+    {
+        $recommendations = $this->recommendationsRepository->getByRecommendedEvent($jsonDocument->getId());
+
+        if ($recommendations->isEmpty()) {
+            return $jsonDocument;
+        }
+
+        return $jsonDocument->applyAssoc(
+            function (array $body) use ($recommendations) {
+                $body['metadata']['recommendationFor'] = $this->recommendationsToArray($recommendations);
```

Comment:
Should this also become `recommendationsFor` then?

Reply:
Ok, now I'm also getting a bit confused 😄

But I think the rule is:
- Recommendations is always plural
- RecommendationFor is always singular

The code should reflect this now.

Reply:
With an example:
- Event A has as **recommendations** event 1 and event 2
- Event B has as **recommendations** event 1 and event 3
- Event 1 is a **recommendationFor** event A and event B